include/SDL3/SDL_clipboard.h

```diff
@@ -129,6 +129,53 @@ extern DECLSPEC char * SDLCALL SDL_GetPrimarySelectionText(void);
  */
 extern DECLSPEC SDL_bool SDLCALL SDL_HasPrimarySelectionText(void);
 
+/**
+ * Set clipboard data for the provided mime type
+ *
+ * \param data      The data buffer to read from
+ * \param len       The length of the data
+ * \param mime_type The mime type of the data
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_GetClipboardData
+ * \sa SDL_HasClipboardData
+ */
+extern DECLSPEC int SDLCALL SDL_SetClipboardData(void *data, size_t len, const char *mime_type);
```

Comment:
I don't think this approach will work because I'm assuming the OS will clear the previous "offer" every time this is called so this doesn't allow for offering multiple mime-types at once. I'm still thinking what's best here. Suggestions are welcome.